diesel_cli/src/migrations/diff_schema.rs

```diff
@@ -0,0 +1,596 @@
+                let mut qb = diesel::pg::PgQueryBuilder::default();
+                diff.generate_up_sql(&mut qb)?;
+                qb.finish()
+            }
+            #[cfg(feature = "sqlite")]
+            InferConnection::Sqlite(_) => {
+                let mut qb = diesel::sqlite::SqliteQueryBuilder::default();
+                diff.generate_up_sql(&mut qb)?;
+                qb.finish()
+            }
+            #[cfg(feature = "mysql")]
+            InferConnection::Mysql(_) => {
+                let mut qb = diesel::mysql::MysqlQueryBuilder::default();
+                diff.generate_up_sql(&mut qb)?;
+                qb.finish()
+            }
+        };
+
+        let down = match conn {
+            #[cfg(feature = "postgres")]
+            InferConnection::Pg(_) => {
+                let mut qb = diesel::pg::PgQueryBuilder::default();
+                diff.generate_down_sql(&mut qb)?;
+                qb.finish()
+            }
+            #[cfg(feature = "sqlite")]
+            InferConnection::Sqlite(_) => {
+                let mut qb = diesel::sqlite::SqliteQueryBuilder::default();
+                diff.generate_down_sql(&mut qb)?;
+                qb.finish()
+            }
+            #[cfg(feature = "mysql")]
+            InferConnection::Mysql(_) => {
+                let mut qb = diesel::mysql::MysqlQueryBuilder::default();
+                diff.generate_down_sql(&mut qb)?;
+                qb.finish()
+            }
+        };
+        up_sql += &up;
+        up_sql += "\n";
+        down_sql += &down;
+        down_sql += "\n";
+    }
+
+    Ok((up_sql, down_sql))
+}
+
+fn is_same_type(ty: &ColumnType, tpe: ColumnType) -> bool {
+    if ty.is_array != tpe.is_array
+        || ty.is_nullable != tpe.is_nullable
+        || ty.is_unsigned != tpe.is_unsigned
+    {
+        return false;
+    }
+
+    let mut is_same_schema = ty.schema == tpe.schema;
+    if !is_same_schema
+        && ((ty.schema.as_deref() == Some("pg_catalog") && tpe.schema.is_none())
+            || (tpe.schema.as_deref() == Some("pg_catalog") && ty.schema.is_none()))
+    {
+        is_same_schema = true;
+    }
+
+    if ty.sql_name.to_lowercase() == tpe.sql_name.to_lowercase() && is_same_schema {
+        return true;
+    }
+    if !is_same_schema {
+        return false;
+    }
+    let compatible_types = compatible_type_list();
+
+    if let Some(compatible) = compatible_types.get(&ty.sql_name.to_lowercase() as &str) {
+        return compatible.contains(&((&tpe.sql_name.to_lowercase()) as &str));
+    }
+    if let Some(compatible) = compatible_types.get(&tpe.sql_name.to_lowercase() as &str) {
+        return compatible.contains(&(&ty.sql_name.to_lowercase() as &str));
+    }
+    false
+}
+
+#[allow(clippy::enum_variant_names)]
+enum SchemaDiff {
+    DropTable {
+        table: TableName,
+        columns: TableData,
+        foreign_keys: Vec<ForeignKeyConstraint>,
+    },
+    CreateTable {
+        to_ceate: TableDecl,
```

Comment:
Probably
```suggestion
        to_create: TableDecl,
```

Reply:
Good point :+1:
(Fixed in one of the force pushes)